x-pack/filebeat/input/httpjson/request.go

```diff
@@ -37,8 +37,17 @@ func (c *httpClient) do(stdCtx context.Context, trCtx *transformContext, req *ht
 	if err != nil {
 		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
 	}
+	defer resp.Body.Close()
+
+	// Read the whole resp.Body so we can release the conneciton.
+	// This implementaion is inspired by httputil.DumpResponse
+	resp.Body, err = drainBody(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
 	if resp.StatusCode > 399 {
-		body, _ := ioutil.ReadAll(resp.Body)
+		body, _ := io.ReadAll(resp.Body)
 		resp.Body.Close()
```

Comment:
With the new `defer resp.Body.Close()` above this will close the body twice.

Reply:
Probably safe, but it is easy to just rely on the deferred call entirely.

Reply:
Good catch! I removed it on ed127f1ba4